### ISSUE TITLE
no-unused-expressions false positive warning

### ISSUE THREAD

**Issue by JustFly1984** [eslint/eslint#7918]
**Tell us about your environment**

* **ESLint Version: 3.8.1**
* **Node Version: 7.4.0**
* **npm Version: 4.0.3**

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**What did you do? Please include the actual source code causing the issue.**
  I have this peace of code in my project, and I do not understand, why Eslint showing a warning ```no-unused-expressions``` .

I set ```// eslint-disable-line no-unused-expressions``` on the line which causes warning

```
body.append('model-file', this.fileInput.files[0])

 body.append('language', this.props.language) // eslint-disable-line no-unused-expressions

 body.append('sampleRate', this.props.sampleRate)
```
**What did you expect to happen?**
 no warnings
**What actually happened? Please include the actual, raw output from ESLint.**

Warning in ./src/components/index/recording-controls.js

/spa/src/components/index/recording-controls.js
  92:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions

✖ 1 problem (0 errors, 1 warning)

You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.

**Comment by platinumazure** [comment]
That's puzzling. Could you please include the whole file (maybe in a Gist), in case the issue is really caused by a line somewhere else which confuses the parser?

I can't reproduce this issue in the [ESLint demo](http://eslint.org/demo/), even when the `// eslint-disable-line` comment is removed.

**Comment by JustFly1984** [comment]
```
import {
  React, // eslint-disable-line no-unused-vars
  Component,
  PropTypes
} from '../../nmi'

import './recording-controls.css'

import latencyChart from '../../charts/latency-chart'

import keywordChart from '../../charts/keyword-chart'

import sentimentChart from '../../charts/sentiment-chart'

class RecordingControls extends Component {
  static propTypes = {
    closeWebMedia: PropTypes.func.isRequired,
    openStream: PropTypes.func.isRequired,
    closeStream: PropTypes.func.isRequired,
    startRecordingButtonTitle: PropTypes.string.isRequired,
    language: PropTypes.string.isRequired,
    sampleRate: PropTypes.string.isRequired,
    changeStartRecordingButtonTitle: PropTypes.func.isRequired
  }

  state = {
    modelTypeIndicator: 'default',
    origFileNameIndicator: 'none',
    startDisabled: false,
    stopDisabled: true,
    uploadDisabled: false,
    resetDisabled: false
  }

  start = event => {
    console.log('start')
    this.setState({
      startDisabled: true,
      stopDisabled: false,
      uploadDisabled: true,
      resetDisabled: true
    })

    this.props.openStream()

    this.props.changeStartRecordingButtonTitle('Loading Engine ...')
    // TODO: Charts
    latencyChart.reset()
    keywordChart.reset()
    sentimentChart.reset()
  }

  stop = () => {
    console.log('stop')

    this.setState({
      startDisabled: false,
      stopDisabled: true,
      uploadDisabled: false,
      resetDisabled: false
    })

    this.props.closeStream()

    this.props.changeStartRecordingButtonTitle('Start Recording')

    this.props.closeWebMedia()
  }

  upload = () => {
    this.fileInput.click()
  }

  fileInputHandle = () => {
    this.setState({
      modelTypeIndicator: 'the model is being built...'
    })

    this.setState({
      startDisabled: true,
      stopDisabled: true,
      uploadDisabled: true,
      resetDisabled: true
    })

    let body = new FormData()

    console.log(`this.fileInput: `, this.fileInput)

    body.append('model-file', this.fileInput.files[0])

    body.append('language', this.props.language) // eslint-disable-line no-unused-expressions

 +  body.append('sampleRate', this.props.sampleRate)

    fetch(
      'https://localhost:9191/model/upload',
      {
        method: 'POST',
        credentials: 'include',
        mode: 'cors',
        body
      })
    .then(res => res.json())
    .then(json => {
      console.log(`recording-controls fileInputHandle fetch json: ${json}`, json)

      if (json.result === 'ok') {
        this.setState({
          modelTypeIndicator: json.modelType,
          origFileNameIndicator: json.origFilename,
          startDisabled: false,
          stopDisabled: true,
          uploadDisabled: false,
          resetDisabled: false
        })
      }
    })
    .catch(err => {
      console.error(`recording-controls fileInputHandle fetch err: ${err}`, err)
    })
  }

  reset = () => {
    this.setState({
      startDisabled: true,
      stopDisabled: true,
      uploadDisabled: true,
      resetDisabled: true
    })

    fetch('https://localhost:9191/model/reset', {
      method: 'post',
      mode: 'cors',
      credentials: 'include'
    }).then(res => {
      return res.json()
    }).then(json => {
      console.log(`recording-controls reset button fetch json: ${json}`, json)

      if (json.result === 'ok') {
        this.setState({
          modelTypeIndicator: json.modelType,
          origFileNameIndicator: json.origFilename,
          startDisabled: false,
          stopDisabled: true,
          uploadDisabled: false,
          resetDisabled: false
        })
      }
    }).catch(err => {
      console.error(`recording-controls reset button fetch err: ${err}`, err)
    })
  }

  render = () => {
    return (
      <div className='row'>
        <div className='col-sm-12'>
          <button
            className='btn btn-primary'
            onClick={this.start}
            disabled={this.state.startDisabled}
          >
            {this.props.startRecordingButtonTitle}
          </button>

          <button
            className='btn btn-primary'
            onClick={this.stop}
            disabled={this.state.stopDisabled}
          >
            Stop Recording
          </button>

          <button
            className='btn btn-primary'
            onClick={this.upload}
            disabled={this.state.uploadDisabled}
          >
            Set custom model/vocabulary
          </button>

          <button
            className='btn btn-primary'
            onClick={this.reset}
            disabled={this.state.resetDisabled}
          >
            Reset custom model
          </button>

          <form>
            <div style={{ height: '0', overflow: 'hidden' }}>
              <input
                ref={input => { this.fileInput = input }}
                onChange={this.fileInputHandle}
                type='file'
              />
            </div>
          </form>
        </div>
        <div className='col-sm-12'>
          <span>Model Type: </span>

          <span>
            {this.state.modelTypeIndicator}
          </span>

          <span>, File Name: </span>

          <span>
            {this.state.origFileNameIndicator}
          </span>
        </div>
      </div>
    )
  }
}

export default RecordingControls

```

**Comment by JustFly1984** [comment]
Please confirm that you copied this code. I will clean my comment with code after your confirmation

**Comment by JustFly1984** [comment]
PS: eslint-loader@1.6.0

**Comment by platinumazure** [comment]
@JustFly1984 What's that plus doing on the line after the lint message? At first I thought it was a diff/patch thing, but if that's actually in the code, then the code actually parses to

```js
body.append('language', this.props.language) + body.append('sampleRate', this.props.sampleRate)
```

Which is indeed an unused expression. If you get rid of the plus, you're probably good. (Or, if you meant to do unary plus, you need a semicolon on the previous line.)

**Comment by JustFly1984** [comment]
Wow, missed that! Thank you very much!

**Comment by platinumazure** [comment]
Glad to help!
